include/ProgramRepresentation/FullFile.h

```diff
@@ -3,8 +3,26 @@
 
 #include "ProgramRepresentation/ProgramFunction.h"
 
-// reflects a file. this class manages all the functions that make up a single
-// file.
+/* reflects a file. this class manages all the functions that make up a single
+file. how everything in `ProgramRepresentation` is structured is as follows:
+
+- a `FullFile` is composed of functions (`ProgramFunction` s) that the user declares.
+- `ProgramFunction` s are composed of LLVM IR branches (`ProgramPoints` s)
+- and these branches contain information mapping each variable (and any of its aliases)
+to a set of methods called on them, which are the `DisjointPVAliasSets`.
+- these disjoint sets are composed of `PVAliasSet` s, which holds a set of program variables
+(typed `ProgramVariable`) which are must-aliases of each other, and the set of methods
+called on them (typed `MethodSet`)
+- `ProgramVariable` s are thus primitive objects that are isomorphic to the
```

Comment:
What does "primitive" mean in this context? That makes me think of e.g., Java's distinction between "primitive" (unboxed) types like `int` and object types. I don't think that's the intended meaning here; these are more like the "leaves" of the structure that you're describing here.